src/library/module.cpp

```diff
@@ -49,17 +49,29 @@ namespace lean {
 struct olean_header {
     // 5 bytes: magic number
     char marker[5] = {'o', 'l', 'e', 'a', 'n'};
-    // 1 byte: version, currently always `1`
-    uint8_t version = 1;
-    // 42 bytes: build githash, padded with `\0` to the right
-    char githash[42];
+    // 1 byte: version, incremented on structural changes to header
+    uint8_t version = 2;
+    // 1 byte of flags:
+    // - whether persisted bignums use GMP or Lean-native encoding
```

Comment:
```suggestion
    // * bit 0: whether persisted bignums use GMP or Lean-native encoding
    // * bit 1-7: reserved
```

Reply:
Could potentially use
```
uint8_t uses_gmp : 1; 
uint8_t reserved_flags : 7;
```
instead

Reply:
I'd prefer if it didn't, this adds an unnecessary dependency on [more C wibbly bits](https://stackoverflow.com/a/19376742/890016) than necessary

Reply:
Ouch, I retract that suggestion then.